sdk/python/kfp/client/client.py

```diff
@@ -1373,7 +1373,7 @@ def upload_pipeline(
         """
         if pipeline_name is None:
             pipeline_name = os.path.splitext(
-                os.path.basename('something/file.txt'))[0]
+                os.path.basename(pipeline_package_path))[0]
```

Comment:
Sounds good to me 👍🏼

Reply:
@connor-mccarthy, I looked into this implementation further, and I think the backend is enforcing setting the name of the pipeline to the package name when there is no name provided https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/pipeline_upload_server.go#L113  https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/util.go#L63 and the IR spec is parsed later on https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/resource/resource_manager.go#L264. It seems like the best long-term solution would be to have the upload pipeline backend use the IR spec name if the uploaded file is an IR spec and no name query param is provided. However, since the api v2 is still in active progress, would it make sense for me to have the sdk parse the yaml spec and send the name to the api request call for now? The api changes seem pretty straightforward and I'm happy to do them, but wanted to ask first in case this implementation would add complications to larger api changes in progress.

Reply:
> However, since the api v2 is still in active progress, would it make sense for me to have the sdk parse the yaml spec and send the name to the api request call for now?

Agreed on all of that. Thanks for being mindful of the BE development. The KFP SDK implementation is a good short term option and is preferred over the current SDK implementation.

However, I agree that in the long run this should live on the BE, otherwise SDK requests contain duplicate information (i.e., creates coupling) that need to be maintained.

> The api changes seem pretty straightforward and I'm happy to do them, but wanted to ask first in case this implementation would add complications to larger api changes in progress.

If it is easy and non-disruptive to add these changes to the BE, let's do that. If it's disruptive, let's start with the SDK changes.

@gkcalat, would it be disruptive for @tarat44 to add the following logic to the BE's upload pipeline handling?

BE accepts a null value in the request's `name` field and, when this happens, the BE will extract and use the pipeline name in the PipelineSpec message as the name

Reply:
@connor-mccarthy @gkcalat, I went ahead with making the changes to the sdk for now but happy to pivot to the backend if desired.

Reply:
Hi @tarat44 and @connor-mccarthy! Thank you for your contribution!
SGTM on implementing this feature in the backend.

The name of an uploaded pipeline is set [here](https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/server/pipeline_upload_server.go#L113). Feel free to propose a change. Please, make sure to adjust the unit tests if we have any that check the old naming logic. Once ready, please ping us.

Note, as we are actively working on making 2.0.0-beta.0 release, your PR will probably be merged after the release. And you may need to rebase your change if you create it before the release.

Reply:
Sounds good, thank you for the guidance